prime_sieve.py

```
@@ -0,0 +1,66 @@
+"""Prime sieve implementation capable of generating primes up to 1 billion.
+
+This module provides a segmented sieve algorithm which can be used to
+iterate over prime numbers up to a specified limit. The segmented
+approach avoids allocating a huge contiguous array, keeping memory
+usage reasonable even for very large limits such as one billion.
+"""
+
+from math import isqrt
+
+
+def simple_sieve(limit):
```

Ruff (0.11.9)

Missing return type annotation for public function `simple_sieve`

(ANN201)